### PR TITLE
PT-13152 - Add turnover report

### PR DESCRIPTION
**Pull request by swDennis**

### CODE REVIEW

File: Components/TransactionReport/TransactionReport.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            ->execute()
+            ->fetchAll(PDO::FETCH_GROUP);
+
+        return new ReportResult($result);
+    }
+
+    /**
+     * @return array<int, int>
+     */
+    private function getPaymentIds()
```
Comment (mitelg) [inline Components/TransactionReport/TransactionReport.php:126]: why not using the `\SwagPaymentPayPalUnified\Components\PaymentMethodProvider`?

File: Components/TransactionReport/TransactionReport.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    /**
+     * @param string $shopwareVersion
+     *
+     * @return void
+     */
+    public function report($shopwareVersion, Client $client)
+    {
+        $reportResult = $this->getReportResult($this->getReportedOrderIds());
```
Comment (mitelg) [inline Components/TransactionReport/TransactionReport.php:56]: why not creating the client in this method? I guess you do not want to add those method parameters to the constructor, as they are not used by every public method?

File: Tests/Functional/Components/TransactionReport/TransactionReportTest.php [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        ];
+
+        $expectedTurnoverResult = [
+            'EUR' => 14442.3,
+            'DKK' => 11962.28,
+            'CHF' => 12478.41,
+            'USD' => 15057.28,
+        ];
+
+        /** @var ReportResult $result */
```
Comment (mitelg) [inline Tests/Functional/Components/TransactionReport/TransactionReportTest.php:179]: do not use var annotations in tests. assert, if necessary to know the type

**Review by mitelg — CHANGES_REQUESTED** [review]

File: Components/TransactionReport/TransactionReport.php (hunk 2 above)
Comment (swDennis) [reply Components/TransactionReport/TransactionReport.php:55]: for testin

File: Components/TransactionReport/TransactionReport.php (hunk 1 above)
Comment (swDennis) [reply Components/TransactionReport/TransactionReport.php:125]: Unnecessary dependency

File: Resources/services/subscribers.xml [hunk 4]
```diff
@@ -328,5 +328,13 @@
                  public="true">
             <tag name="shopware.event_subscriber"/>
         </service>
+
+        <service id="paypal_unified.subscriber.transaction_report"
+                 class="SwagPaymentPayPalUnified\Subscriber\TransactionReportSubscriber"
+                 public="true">
+            <argument type="service" id="dbal_connection"/>
+            <argument type="service" id="Shopware\Components\ShopwareReleaseStruct"/>
```
Comment (mitelg) [inline Resources/services/subscribers.xml:336]: this will not work in older SW versions, right?

Comment (swDennis) [reply Resources/services/subscribers.xml]: is already fixed...

**Review by mitelg — APPROVED** [review]

File: Subscriber/TransactionReportSubscriber.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        $shopwareVersion = '';
+        if (\defined('\Shopware::VERSION')) {
+            $shopwareVersion = Shopware::VERSION;
+        } else {
+            $shopwareVersion = $this->container->get('shopware.release.version');
+        }
+
+        (new TransactionReport($this->connection))->report(
+            Shopware::VERSION,
```
Comment (mitelg) [inline Subscriber/TransactionReportSubscriber.php:56]:
```suggestion
            $shopwareVersion,
```

**Pull request merged**